tests/20-rbac-test.py

```diff
@@ -0,0 +1,166 @@
+            bundle_yaml = stream.read()
+        bundle = yaml.safe_load(bundle_yaml)
+        cls.deployment.load(bundle)
+
+        # Allow some time for Juju to provision and deploy the bundle.
+        cls.deployment.setup(timeout=SECONDS_TO_WAIT)
+
+        # Wait for the system to settle down.
+        application_messages = {'kubernetes-worker':
+                                'Kubernetes worker running.'}
+        cls.deployment.sentry.wait_for_messages(application_messages,
+                                                timeout=900)
+
+    @property
+    def easyrsas(self):
+        return self.deployment.sentry['easyrsa']
+
+    @property
+    def etcds(self):
+        return self.deployment.sentry['etcd']
+
+    @property
+    def flannels(self):
+        return self.deployment.sentry['flannel']
+
+    @property
+    def loadbalancers(self):
+        return self.deployment.sentry['kubeapi-load-balancer']
+
+    @property
+    def masters(self):
+        return self.deployment.sentry['kubernetes-master']
+
+    @property
+    def workers(self):
+        return self.deployment.sentry['kubernetes-worker']
+
+    def test_master_services(self):
+        '''Test if the master services are running.'''
+        for master in self.masters:
+            services = [
+                'snap.kube-apiserver.daemon',
+                'snap.kube-controller-manager.daemon',
+                'snap.kube-scheduler.daemon'
+            ]
+            for service in services:
+                self.assertTrue(check_systemd_service(master, service))
+
+    def test_worker_services(self):
+        '''Test if the worker services are running.'''
+        for worker in self.workers:
+            services = [
+                'docker',
+                'snap.kubelet.daemon',
+                'snap.kube-proxy.daemon'
+            ]
+            for service in services:
+                self.assertTrue(check_systemd_service(worker, service))
+
+    def test_kubeconfig(self):
+        '''Test that the kubeconfig exists so that kubectl commands can run.'''
+        # typical client configs
+        for master in self.masters:
+            output, rc = master.run('grep server: /home/ubuntu/config')
+            self.assertTrue(rc == 0)
+
+        # typical client configs
+        for worker in self.workers:
+            ubuntu, rc = worker.run('grep server: /home/ubuntu/.kube/config')
+            self.assertTrue(rc == 0)
+            root, rc = worker.run('grep server: /root/.kube/config')
+            self.assertTrue(rc == 0)
+            self.assertTrue(ubuntu == root)
+
+        # post rbac component configs
+        for worker in self.workers:
+            kubelet, rc = worker.run('grep token: /root/cdk/kubeletconfig')
+            self.assertTrue(rc == 0)
+            proxy, rc = worker.run('grep token: /root/cdk/kubeproxyconfig')
+            self.assertTrue(rc == 0)
+            self.assertTrue(kubelet != proxy)
+
+    def test_cluster_info(self):
```

Comment:
A lot of these tests look to be duplicates from 20-charm-validation.py. Mind removing the duplicates?